client/offchain/src/api/http.rs

```diff
@@ -688,31 +688,38 @@ impl fmt::Debug for HttpWorkerRequest {
 
 #[cfg(test)]
 mod tests {
+	use core::convert::Infallible;
 	use crate::api::timestamp;
 	use super::http;
-	use futures::prelude::*;
-	use futures01::Future as _;
 	use sp_core::offchain::{HttpError, HttpRequestId, HttpRequestStatus, Duration};
 
 	// Returns an `HttpApi` whose worker is ran in the background, and a `SocketAddr` to an HTTP
 	// server that runs in the background as well.
 	macro_rules! build_api_server {
 		() => {{
+			macro_rules! tokio_run {
```

Comment:
Not sure if the number of repetitions is worth a macro. We could either inline it or just use a closure?

Reply:
We can't use closures since the two futures differ in concrete types (this would require a generic closure, it's what I tried originally) - I can inline it or use a helper function :+1:

Reply:
Used a generic helper function.